sway-lib-std/src/vec.sw

```diff
@@ -141,4 +142,71 @@ impl<T> Vec<T> {
     pub fn is_empty(self) -> bool {
         self.len == 0
     }
+
+    /// Removes and returns the element at position `index` within the vector,
+    /// shifting all elements after it to the left.
+    /// Panics if `index >= self.len`
+    pub fn remove(mut self, index: u64) -> T {
+        assert(index < self.len);
+
+        let val_size = size_of::<T>();
+        let buf_start = self.buf.ptr();
+        let mut ptr = buf_start + val_size * index;
+
+        // Read from `ptr`
+        let ret = read(ptr);
+
+        // Shift everything down to fill in that spot.
+        let end = buf_start + val_size * self.len;
+        while ptr < end {
+            copy(ptr, ptr + val_size, val_size);
+            ptr += val_size;
+        }
+
+        // Decrease length.
+        self.len -= 1;
+        ret
+    }
+
+    /// Inserts an element at position `index` within the vector, shifting all
+    /// elements after it to the right.
+    /// Panics if `index > len`.
+    pub fn insert(mut self, index: u64, element: T) {
+        assert(index <= self.len);
+
+        // If there is insufficient capacity, grow the buffer.
+        if self.len == self.buf.cap {
+            self.buf.grow();
+        }
+
+        let val_size = size_of::<T>();
+        let buf_start = self.buf.ptr();
+
+        // The spot to put the new value
+        let index_ptr = buf_start + index * val_size;
+
+        // Shift everything over to make space. 
+        let mut curr_ptr = buf_start + self.len * val_size;
+        while curr_ptr > index_ptr {
+            copy(curr_ptr, curr_ptr - val_size, val_size);
+            curr_ptr -= val_size;
+        }
+
+        // Write `element` at pointer `index`
+        write(index_ptr, element);
+
+        // Increment length.
+        self.len += 1;
+    }
+
+    /// Removes the last element from a vector and returns it, or [`None`] if it
+    /// is empty.
+    fn pop(mut self) -> Option<T> {
+        if self.len == 0 {
+            return Option::None;        
+        } else {
```

Comment:
You do not need to use the `else` here unless the compiler hasn't reached the point of development where it understands that is okay. You should be able to continue to control flow as usual without `else`.
If there are no complaints from the compiler then can you remove the "else" so that there is less spaghetti?

Reply:
My preference is to remove as many indents as possible. I've seen the horrors that it could lead to so I'm generally put off when I see them and cannot get around them easily. 
Following rust is fine but if they write spaghetti then it shouldn't be followed imo